src/content/docs/en/basics/layouts.mdx

```diff
@@ -258,6 +258,52 @@ const { title } = Astro.props.frontmatter || Astro.props;
 </html>
 ```
 
+### Working with TypeScript
+
+Any astro layout can be modified to introduce typesafety & autocompletion by providing the types for your props:
+
+```astro ins={2-7} title="src/components/MyLayout.astro"
+---
+interface Props { 
+	title: string;
+  description: string;
+  publishDate: Date;
+  viewCount: number;
+}
+const { title, description, publishDate, viewCount } = Astro.props;
+---
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="description" content={description}>
+    <title>{title}</title>
+  </head>
+  <body>
+    <header>
+      <p>Published on {publishDate}</p>
+      <p>Viewed by {viewCount} folks</p>
+    </header>
+    <main>
+      <slot />
+    </main>
+  </body>
+</html>
+```
+
+You can override Astro's default `Props` to handle situations where your layout recieves props of different types:
```

Comment:
⚠️ friction alert ⚠️ `You can override ... different types` is incorrect afaik because `Props` can be defined/override to include different types as described in snippet above and I can't think of what it should be 😓
Can any experienced TypeScript developers confirm suggest a better approach? Thanks in advance ❤️